src/include/index/ivf_pq_index.h

```diff
@@ -1537,6 +1546,13 @@ class ivf_pq_index {
     if (num_clusters_ != rhs.num_clusters_) {
       return false;
     }
+    if (distance_metric_ != rhs.distance_metric_) {
+      std::cout << "distance_metric_ != rhs.distance_metric_"
+                << static_cast<int>(distance_metric_)
+                << " != " << static_cast<int>(rhs.distance_metric_)
+                << std::endl;
```

Comment:
(Minor) I'm happy with adding printing here, but if we do, let's do it for all the cases above as well. So could be a separate small PR to improve this if you wanted (or it can be in this one). Same with the Vamana code below.